packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -197,6 +197,7 @@ export default defineRenderHandler(async (event) => {
   const isRenderingPayload = PAYLOAD_URL_RE.test(url)
   if (isRenderingPayload) {
     url = url.substring(0, url.lastIndexOf('/')) || '/'
+    url += !url.endsWith("/") ? "/" : "";
```

Comment:
How about instead we strip the trailing slash at L257?

Reply:
@danielroe it works if I do it on L266, but we might need to do it on L257 too. If this is needed on both lines, I'm considering adding a small function like this to remove trailing slash, but not for the main "/" path.

    const removeTrailingSlash = (url2) => url2.endsWith('/') && url2 != '/' ? url2.substring(0, url2.length - 1) : url2

Should I change this? Otherwise I will just replace these double quotes and remove the semicolon to make linting happy.

Reply:
No L266 is the right place. We can use `withoutTrailingSlash` from `ufo` in this place, because it will only be used in the prerenderer, not added to the runtime server bundle.

Reply:
@danielroe looks clean and works nicely, thanks for the suggestion 👍